client/blocks/product-selector/index.jsx

```diff
@@ -143,10 +144,21 @@ export class ProductSelector extends Component {
 		} );
 	}
 
-	handleCheckoutForProduct = productObject => {
-		const { selectedSiteSlug } = this.props;
+	handleCheckoutForProduct = ( productObject, product ) => {
+		const { currentPlanSlug, selectedSiteSlug } = this.props;
+
+		const tracksEventProps = {
+			upgrading_to: productObject.product_slug,
+			current_plan: currentPlanSlug,
+		};
+
+		const productPurchase = this.getPurchaseByProduct( product );
+		if ( productPurchase && productPurchase.productSlug ) {
+			tracksEventProps.current_product = productPurchase.productSlug;
```

Comment:
I'd generally advise against mutating objects anywhere and for any reason when that's possible. Could we construct a new object and pass it to `recordTracksEvent` instead?